### PR TITLE
Landingpage + Website to register user

### PR DESCRIPTION
**Pull request by Zylesto**
- landingpage added
- website to register user added

### CODE REVIEW

File: js/register_user_website/validate_input.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    for (var i = 0; i < alerts_text.length; i++) {
+        if (alerts_text[i].textContent == warning_text) {
+            return;
+        }
+    }
+    for (var i = 0; i < alerts_text.length; i++) {
+        if (alerts_text[i].textContent == "") {
+            alerts_text[i].textContent = warning_text;
+            alerts_text[i].parentElement.parentElement.setAttribute("style", "visibility: visible !important;");
+            break;
```
Comment (ordass00) [inline js/register_user_website/validate_input.js:46]: gehört der teil nicht in den else teil von der oberen for schleife? Dann braucht man nicht 2 mal die for schleife

**Review by ordass00 — APPROVED** [review]

File: js/register_user_website/validate_input.js (hunk 1 above)
Comment (Zylesto) [reply js/register_user_website/validate_input.js]: Habe es umgeschrieben, gute Idee 🥇

File: php/register_account_details/register_account_details.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        case "check_duplicates_email":
+            $user = get_user_by_mail($conn, $reqObj->email);
+            if($user == false){
+                echo json_encode(["error" => false, "duplicate" => false]);
+            }
+            else if($user != false){
+                echo json_encode(["error" => false, "duplicate" => true]);
+            }
+            else if($user == null){
```
Comment (ArminShinobi) [inline php/register_account_details/register_account_details.php:24]: return $user == null ? json_encode(["error" => false, "duplicate" => false]) : json_encode(["error" => true, "duplicate" => true]);
Perhaps error true or false and not duplicate but an appropriate message (for the toast)?

File: php/register_account_details/register_account_details.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                echo json_encode(["error" => false, "duplicate" => false]);
+            }
+            else
+            {
+                echo json_encode(["error" => true, "duplicate" => null]);
+            }
+        break;
+        case "check_duplicates_username":
+            $user = get_user_by_username($conn, $reqObj->username);
+            if($user == false){
```
Comment (ArminShinobi) [inline php/register_account_details/register_account_details.php:34]: Please see above.

File: php/register_account_details/register_account_details.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+$request = file_get_contents("php://input");
+
+if(isset($request) && !empty($request))
+{
+    $reqObj = json_decode($request);
+    //Checking for duplicates depending on input
+    switch($reqObj->method)
+    {
+        case "check_duplicates_email":
```
Comment (ArminShinobi) [inline php/register_account_details/register_account_details.php:16]: I would have several methods for the both duplicate checks and the registration.

File: html/register_account_details.html [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  <main class="d-flex align-items-center justify-content-center">
+    <div class="personal-details-input">
+      <div class="d-flex align-items-center justify-content-center">
+        <h1 class="m-2">Personal Data</h1>
+      </div>
+      <p class="m-2">To get started, we need some information about you.</p>
+      <form id="f1">
+        <div class="d-flex align-items-center justify-content-center">
+          <div class="form-group">
+            <input id="username_input" type="text" class="form-control m-2" placeholder="Username" name="username" autocomplete="username">
```
[the remark this answers is not in the document]
Comment (ArminShinobi) [reply html/register_account_details.html:42]: https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation

File: html/register_account_details.html [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    <title>Register</title>
+  </head>
+
+  <body>
+    <div class="register_image"></div>
+    <div
+      aria-live="polite"
+      aria-atomic="true"
+      class="d-flex justify-content-center align-items-center m-2 toasts"
+    >
```
Comment (ordass00) [inline html/register_account_details.html:24]: please use consistent formating, e.g like this
`<div aria-live="polite" aria-atomic="true" class="d-flex justify-content-center align-items-center m-2 toasts">`

File: css/register_account_details.css [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+.show-hide-pwd-eye:active {
+  padding: 8px 10px 5px 10px;
+}
+
+.align-center {
+  position: relative;
+  align-items: center;
+  left: 25px;
+  top: 15px;
+}
```
Comment (ordass00) [inline css/register_account_details.css:76]: There are a lot of duplications with my css code. Maybe we should discuss about creating a global css file for things everyone uses quite often?

**Review by ordass00 — CHANGES_REQUESTED** [review]

File: html/register_account_details.html [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+    <script
+      src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta2/dist/js/bootstrap.bundle.min.js"
+      integrity="sha384-b5kHyXgcpbZJO/tY9Ul7kGkf1S0CWuKcCD38l8YkeH8z8QjE0GmW1gYU5S9FOnJ0"
+      crossorigin="anonymous"
+    ></script>
+    <script src="../js/register_user_website/validate_input.js"></script>
+    <script src="../js/register_user_website/register_user.js"></script>
+  </body>
+</html>
```
Comment (ordass00) [inline html/register_account_details.html:139]: please use consistent formating, e.g like this
`<div aria-live="polite" aria-atomic="true" class="d-flex justify-content-center align-items-center m-2 toasts">`

File: html/register_account_details.html [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        /><br />
+        <input
+          type="email"
+          class="form-control"
+          placeholder="name@example.com"
+          id="email_input"
+          required
+          autocomplete="email"
+          maxlength="45"
+        /><br />
```
Comment (ArminShinobi) [inline html/register_account_details.html:109]: <br> shouldn't be used

File: html/register_account_details.html (hunk 8 above)
Comment (Zylesto) [reply html/register_account_details.html:139]: The formatting of my pages is consistent

**Pull request merged**